internal/server/create_document.go

```diff
@@ -277,9 +313,43 @@ func translateInsertData(selectedTemplateInserts []sirius.Insert, documentTempla
 			}
 		}
 	}
+
+	sort.Slice(documentTemplateInserts, func(i, j int) bool {
+		return documentTemplateInserts[i].Handle < documentTemplateInserts[j].Handle
+	})
+
 	return documentTemplateInserts
 }
 
+func getSortedInsertKeys(selectedTemplateInserts []sirius.Insert) []string {
+	var documentInsertKeys []string
+	for _, in := range selectedTemplateInserts {
+		if !slices.Contains(documentInsertKeys, in.Key) {
+			documentInsertKeys = append(documentInsertKeys, in.Key)
+		}
+	}
+	// insert api response infrequently includes the key all
+	// note insert keys are lowercase so this check is case-sensitive
+	if !slices.Contains(documentInsertKeys, "all") {
+		documentInsertKeys = append(documentInsertKeys, "all")
+	}
+
+	sort.Strings(documentInsertKeys)
+	return documentInsertKeys
+}
+
+func removeDuplicateStr(strSlice []string) []string {
+	allKeys := make(map[string]bool)
```

Comment:
a "fun" trick is you can use `map[string]struct{}` and insert `struct{}{}` values where you don't care about the value, these are empty values so reduces memory use by a byte per item... much performance